src/foam/nanos/approval/AuthenticatedApprovalDAOAuthorizer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.approval',
+  name: 'AuthenticatedApprovalDAOAuthorizer',
+  implements: ['foam.nanos.auth.Authorizer'],
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.*',
+    'foam.nanos.auth.AuthorizationException',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.auth.User',
+    'foam.util.SafetyUtil',
+  ],
+
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+        AuthService authService = (AuthService) x.get("auth");
+        if ( ! authService.check(x, "approval.create") ) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      javaCode: `
+        User user = (User) x.get("user");
+        if ( user == null || ! SafetyUtil.equals(user.getId(), ((ApprovalRequest)obj).getApprover()) ) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnUpdate',
+      javaCode: `
+        ApprovalRequest approvalRequest = (ApprovalRequest) oldObj;
+        User user = (User) x.get("user");
+        Long userId = ((User) x.get("user")).getId();
```

Comment:
```suggestion
        User user = ((Subject) x.get("subject")).getRealUser();
```
And replace userId with user.getId() on line 48.